### ISSUE TITLE
Star mode operators include hidden edges

### ISSUE THREAD

**Issue by KonradHoeffner** [IMISE/snik-cytoscape.js#124]
Ignore hidden edges on star mode operations.

Example:
Uncheck "non-subClassOf", search for "CEO", select bb:ChiefInformationOfficer and do a star. You see nodes that are not connected with visible edges.

**Comment by KonradHoeffner** [comment]
With hidden nodes there is the same problem apparently but it isn't as noticeable though, sometimes its even positive if you uncheck a filter after a star mode operation and instantly get the added edges and nodes.

Problem is to find a selector for "display: element", as ":visible" covers both "display: element" (filters) and "visibility: visible" (star mode), see http://js.cytoscape.org/#selectors/state.

**Comment by KonradHoeffner** [comment]
The reason for the bug is clear now: http://js.cytoscape.org/#style/visibility states "A display: none node hides its connected edges." and I misinterpreted that as the node hiding its connected edges to the cytoscape functions but what was meant was only hiding to the human eye, it is still available over the cytoscape functions.

The possible fix is less clear. We investigated this for over an hour today and didn't find a good solution and may tackle this later.

**Options**

***Remove filtered out nodes instead of setting display:none.***

- Advantage: Small amount of changes for removing and adding in the filter code. Most changes in filter.js and few if any in graph.js.
- Disadvantage: Restoring filters does not work correctly because readded edges fails when they have connected nodes that are still removed. Also, restoring a filter in star mode now incorrectly shows all nodes whose filter turns then visible now.

***Path operations need to exclude "display:none" elements***
- Advantage: No changes to filter.js necessary, only graph.js. 
- Disadvantage: Changes are necessary at too many places in graph.js because there are multiple different path operations. Complicates further enhancements. There does not seem to be a style selector in cytoscape.js.
